Nas work

samiamseid opened the pull request:
Created tool to return list of sessions in NAS that have already been released

samiamseid committed:
Create NAS_tools.py

Started branch

samiamseid committed:
pull candidates for deletion

built tool that returns list of potential candidates for deletion on the NAS storage server

nataliaorlova commented on meso_tools/NAS_tools.py:
```diff
@@ -0,0 +1,96 @@
+    keys = lines[::2]
+    values = lines[1::2]
+    res = {keys[i]: values[i] for i in range(len(keys))}
+    return(res)
```
no need for parenthesis after return

samiamseid replied:
done

nataliaorlova commented on meso_tools/NAS_tools.py:
```diff
@@ -0,0 +1,96 @@
+        query_response (requests.models.Response): Response from the NAS storage containing list of folders and filepaths.  
+        This data is the output of the nas_query function
+    """    
+    import pandas as pd
```
do this in the import section at the top please

samiamseid replied:
done

nataliaorlova reviewed (changes requested):
Generally, I'd make this a class... "nas_api" maybe, where instantiating a class (__init__ method) makes connection and takes in credential file, and other methods can do things with data, for instance, query folder list, query available space, session list, remove folder, create folder, create duplicate, etc. Similar to lims_api class in io_utils.py

nataliaorlova commented on meso_tools/NAS_tools.py:
```diff
@@ -0,0 +1,96 @@
+    return(released_backups, released_backups2)
+            
+def release_check(release_list: str, query_response: requests.models.Response):
+    """Compares the output of a requests query against a list of released sessions, 
```
Let's use Numpy format for docstrings - this is set is autodocstring plugin settings

nataliaorlova commented on meso_tools/NAS_tools.py:
```diff
@@ -0,0 +1,96 @@
+    res = {keys[i]: values[i] for i in range(len(keys))}
+    return(res)
+
+def nas_query(credentials: dict, nas_name: str, release_list: str):
```
add output type using 
def function_name(input1 : str, input2 : int) -> int:
   body of the function
   return output (integer)

samiamseid replied:
done

samiamseid committed:
Update NAStools to Class

reconfigured NAS_tools to be a class

samiamseid committed:
Added delete functions

Added in functions that will delete data. Also updated the documentation

samiamseid commented on meso_tools/NAS_tools.py:
```diff
@@ -0,0 +1,96 @@
+    SYNOLOGY_PASSWORD=x
+
+    Args:
+        cred_path (str): relative string path of txt document containing NAS credentials
```
done

samiamseid committed:
Updated docstring format

changed docstring format to numpy

nataliaorlova commented on meso_tools/NAS_tools.py:
```diff
@@ -0,0 +1,180 @@
+            #save task id for the current deletion jobs
+            task_dict[output.split('/')[-1]]=response.json()['data']['taskid']
+        #store task ID's for later in case we need to stop a job
+        self.task_ids = task_dict
```
add self.task_id to init

samiamseid committed:
Individual Deletion

Changed format to input individual filepaths to queue for deletion.
Deleted files are moved to the recycle bin.
Recycle bin can be deleted in one large job.

samiamseid committed:
Update .gitignore

added vscode settings to gitignore

samiamseid committed:
Merge branch 'master' into NAS_work

Commit:
Merge branch 'master' into NAS_work

samiamseid committed:
Merge branch 'NAS_work' of https://github.com/nataliaorlova/meso_tools into NAS_work

Commit:
script for testing NAS module
